text/0000-dont-trust-builders.md

```diff
@@ -0,0 +1,70 @@
+# Meta
+[meta]: #meta
+- Name: Isolate Registry Credentials from Builder Images
+- Start Date: 2020-01-14
+- CNB Pull Request:
+- CNB Issue:
+- Supersedes: N/A
+
+# Summary
+[summary]: #summary
+
+`pack build` will run `analyze` `restore` and `export` phases in containers built from trusted images rather than the builder image.
+
+# Motivation
+[motivation]: #motivation
+
+Users of `pack` must select a builder image during `pack build`. The builder image is currently used to create all lifecycle phase containers.
+
+When `pack build` is run with the `--publish` flag, the user's registry credentials are injected into the `analyze` and `export` containers as environment variables.
+This means that users must place a high level of trust in their selected builder image.
+
+Users may not realize that credentials are given to builder images and experiment with builders from untrusted vendors.
+
+# What it is
+[what-it-is]: #what-it-is
+
+### Exporter Images
+
+For every release of the lifecycle we will publish an `exporter` image to `docker.io/cnbs/exporter:<version>`,
+where `<version>` is the version of the lifecycle and the image that contains only lifecycle in a scratch image.
+
+When `pack build` executes and is provided the `--publish` flag, it will use the `exporter` image to created the `analyze`, `restore`, and `export` containers.
+
+If the lifecycle version specified in a builder image does not match any known `exporter` image `pack build` will fail
```

Comment:
For the helpful error message, do we want to show people how to create their own lifecycle image or point people to trusting the builder?

Reply:
It would point people towards trusting the builder.

Reply:
Does building an image on the fly help here?

Reply:
I wonder if we should do that and/or list valid versions of lifecycle

Reply:
From slack with @ekcasey:

> I think folks who contribute to lifecycle are the people who will realistically run into this error. If you create a builder with your dev version of the lifecycle, you haven’t made a mistake per se but it can’t run properly w/o being trusted
> we could instead lift valid lifecycles and assume that anyone making their own lifecycle should know what to do?